src/magic_folder/cli.py

```diff
@@ -1032,6 +1022,19 @@ class MagicFolderCommand(BaseOptions):
         "readcaps contained in the master magic-folder directory."
     )
 
+    def parseOptions(self, *args, **kw):
+        """
+        Override usage.Options
```

Comment:
Maybe mention that we're trying to guarantee there is a `node-directory`.

Also, fwiw, maybe there's another way to do this.  We define the `node-directory` option and we define it like:

```
    ["node-directory", "d", None, NODEDIR_HELP]
```

If that were instead

```
    ["node-directory", "d", None, NODEDIR_HELP, FilePath]
```

then I think a value would be required (and then it would be stored as a `FilePath` instead of a `str` or `bytes` or whatever it is now).

Whether you want to make that change or not, the error case here probably needs a new test.

Reply:
Just adding `FilePath` as the factory isn't enough, because that just results in a full-screen exception (instead of a `UsageError`). That's probably a better way to store the path, but I'll leave that to a different PR (because it would be a bunch more changes to everything that uses `node-directory`). I filed https://github.com/LeastAuthority/magic-folder/issues/147

Reply:
Ah, I thought it might still produce a `UsageError` because https://github.com/twisted/twisted/blob/trunk/src/twisted/python/usage.py#L50-L63 but I guess I have misunderstood how that code works or gets used.

Reply:
hmm, weird .. definitely looks like it SHOULD say it's required (at least)

Reply:
I made an SSCCE that also fails to fail: http://paste.debian.net/plain/1153387

Reply:
Hm okay looking more carefully I guess the coercer doesn't get called on the default value, it only gets called on values given as options.